src/write.c

```diff
@@ -1115,11 +1174,20 @@ static avifResult avifEncoderAddImageItems(avifEncoder * encoder,
                                            uint16_t * topLevelItemID)
 {
     const uint32_t cellCount = gridCols * gridRows;
-    const char * infeName = avifIsAlpha(itemCategory) ? infeNameAlpha
+    const char * infeName;
+    if (avifIsAlpha(itemCategory)) {
+        infeName = infeNameAlpha;
 #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
-                            : (itemCategory == AVIF_ITEM_GAIN_MAP) ? infeNameGainMap
+    } else if (itemCategory == AVIF_ITEM_GAIN_MAP) {
+        infeName = infeNameGainMap;
+#endif
+#if defined(AVIF_ENABLE_EXPERIMENTAL_SAMPLE_TRANSFORM)
+    } else if (itemCategory >= AVIF_SAMPLE_TRANSFORM_MIN_CATEGORY && itemCategory <= AVIF_SAMPLE_TRANSFORM_MAX_CATEGORY) {
+        infeName = infeNameSampleTransform;
 #endif
-                                                                   : infeNameColor;
+    } else {
+        infeName = infeNameColor;
+    }
```

Comment:
Nit/optional: If this is implemented in a function, then we can use return statements and avoid the `else`. For example,

```
const char * getInfeName(avifItemCategory itemCategory)
{
    if (avifIsAlpha(itemCategory)) {
        return infeNameAlpha;
    }
#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
    if (itemCategory == AVIF_ITEM_GAIN_MAP) {
        return infeNameGainMap;
    }
#endif
#if defined(AVIF_ENABLE_EXPERIMENTAL_SAMPLE_TRANSFORM)
    ...
#endif
    return infeNameColor;
}
```